src/clients/partitioner_testing.rb

```diff
@@ -20,6 +20,17 @@ def sysfs_name
 end
 
 arg = Yast::WFM.Args.first
-storage = Y2Storage::StorageManager.fake_from_yaml(arg)
-storage.probed.copy(storage.staging)
+case arg
+when /.ya?ml$/
+  storage = Y2Storage::StorageManager.fake_from_yaml(arg)
+  storage.probed.copy(storage.staging)
+when /.xml$/
+  # note: support only xml device graph, not xml output of probing commands
```

Comment:
First of all, this functionality should be yast-storage as `Y2Storage::StorageManager.fake_from_xml`.

That being said, I find this a little bit too low level. I think this should work just fine and is easier to follow:
```ruby
case arg
when /.ya?ml$/
  Y2Storage::StorageManager.fake_from_yaml(arg) # You don't need the second line
when /.xml$/
  storage = Y2Storage::StorageManager.create_test_instance
  # As you know, all those 'y2storage_' will disappear shortly
  storage.probed.load(arg)
  storage.probed.copy(storage.staging)
else
```